src/test/java/it/eng/idsa/dataapp/handler/ContractAgreementMessageHandlerTest.java

```diff
@@ -41,4 +42,13 @@ void handleMessageTest() {
 		assertNull(responseMap.get("payload"));
 		assertTrue(StringUtils.containsIgnoreCase(responseMap.get("header").toString(), message.getId().toString()));
 	}
+
+	@Test
+	void handleMessageHeaderNullTest() {
+
+		responseMap = agreementMessageHandler.handleMessage(null, null);
+
+		assertNotNull(responseMap.get("header"));
```

Comment:
Can we add additional check in case when header message is null?
What kind of message is returned, if it should be returned or exception thrown?